src/core/Table/hooks/useColumnDragAndDrop.tsx

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import {
+  HeaderGroup,
+  Hooks,
+  IdType,
+  makePropGetter,
+  TableInstance,
+  TableKeyedProps,
+  useGetLatest,
+} from 'react-table';
+
+export const useColumnDragAndDrop = <T extends Record<string, unknown>>(
+  isEnabled: boolean,
+) => (hooks: Hooks<T>) => {
+  hooks.getDragAndDropProps = [defaultGetDragAndDropProps(isEnabled)];
+  hooks.useInstance.push(useInstance);
+};
+
+const defaultGetDragAndDropProps = <T extends Record<string, unknown>>(
+  isEnabled: boolean,
+) => (
+  props: TableKeyedProps,
+  {
+    instance,
+    header,
+  }: {
+    instance: TableInstance;
+    header: HeaderGroup;
+  },
+) => {
+  if (!isEnabled || header.disableDragging) {
+    return props;
+  }
+
+  const onDragStart = (event: React.DragEvent<HTMLDivElement>) =>
+    event.dataTransfer.setData('text', header.id);
+
+  const setOnDragColumnStyle = (
+    event: React.DragEvent<HTMLDivElement>,
+    position?: 'left' | 'right',
+  ) => {
+    const columnElement = event.currentTarget as HTMLElement;
+    columnElement.classList.remove('iui-reorder-column-right');
```

Comment:
I think there is a css bug where two borders don't overlap
![2022-03-10 14 11 08](https://user-images.githubusercontent.com/83585998/157659338-acd3e92a-70d8-4cac-86a4-1c61ad905078.gif)

Reply:
So how would you suggest fixing this?

Reply:
I know what is happening here.  The blue stripe is a 2px border that is applied on the left or right.  In this GIF we see the border switching from the left of the ratings column to the right of the quantity column.  We could explore using `::before` and absolute positioning similar to the column resizer.

Reply:
Will be fixed with this https://github.com/iTwin/iTwinUI/pull/518